llama_index/indices/vector_store/retrievers/retriever.py

```diff
@@ -101,9 +98,10 @@ def _retrieve(
             # NOTE: vector store keeps text, returns nodes.
             # Only need to recover image or index nodes from docstore
             for i in range(len(query_result.nodes)):
-                node_id = query_result.nodes[i].get_doc_id()
-                if node_id in self._docstore.docs:
```

Comment:
I think this might actually be the culprit of the slowness. This would load all of the documents from storage into memory.

Reply:
I think we can actually just do the following to avoid the `self._docstore.docs` call:

```python
try:
    query_result.nodes[i] = self._docstore.get_node(node_id)    
except ValueError:
    pass 
```

Reply:
Could you give this a try and see if it fixes the slowness?